_source/_posts/2023-08-30-passwordless-authentication.md

```diff
@@ -0,0 +1,185 @@
+
+{% img blog/passwordless-authentication/ctap.jpg alt:"Graphic of device icons such as phone, laptop, smart card, and smart key along with forms of communication (Bluetooth, USB, and NFC)" width:"800" %}{: .center-image }
+
+W3C introduced a new web standard for web authentication called [WebAuthn](https://www.w3.org/TR/webauthn-2/). A new browser API, the Web Authentication API, extends from the Credential Management API. The Web Authentication API supports secure authentication credentials using asymmetric encryption (where you have a public and private key) instead of passwords. All modern browsers have full or partial WebAuthn capabilities. You can check which browsers and versions support WebAuthn on [caniuse.com](https://caniuse.com/?search=webauthn). The screen capture shows browser support at the time of publishing. 
+
+{% img blog/passwordless-authentication/webauthn-caniuse.jpg alt:"WebAuthn support across browsers showing Internet Explorer and Opera Mini having no support, Firefox desktop and mobile has partial support, and remaining browsers including Chrome, Edge, Safari on desktop and mobile having support." width:"800" %}{: .center-image }
+
+FIDO's Client to Authenticator Protocol complements WebAuthn to create a seamless, easy-to-use, secure authentication experience. 
+
+{% img blog/passwordless-authentication/webauthn-ctap.jpg alt:"Icons representing WebAuthn in the browser and CTAP devices with a heart in between" width:"800" %}{: .center-image }
+
+A new term for this authentication mechanism without passwords is often called "passwordless." And the combination of WebAuthn and CTAP2 hardware devices to enable passwordless authentication is called a "passkey." What a great name!
+
+## FIDO2 authentication with WebAuthn and CTAP
+ 
+So how does this work? At a high level, your FIDO2 authentication flow utilizes each of the three factors:
+1. Something you know, a PIN to unlock the device
+2. Something you have, the authenticator device, which may be a security key or a capable phone
+3. Something you are, biometric data such as fingerprint or facial recognition
+
+Before we authenticate, we need an account with compliant credentials, so let's start with the registration process. We'll use a public demo site demonstrating how passkeys work as we step through what's going on in the registration and authentication process. Follow along at Auth0's [webauthn.me](https://webauthn.me/).
+
+There are a couple of components at play. From the user's perspective, they interact with a website and a device like their phone. From a technology perspective, there's WebAuthn and CTAP2. I'll show screenshots of each step, discuss the interaction between the web app and server of the relying party, and describe the hand-off between WebAuthn and CTAP2.
+
+## Registering a user for passkeys
```

Comment:
I think this section could benefit from some sort of flow diagram. The text makes a great deal of sense, but most people will just skim it in my experience.

If anything, I think these paragraphs could be numbered, each of which referencing a numbered path with a flow diagram demonstrating the interactions between the browser, the authenticator, the authorization server, and the web app on the other end.

The screenshots are certainly useful to demonstrate the user's experience, but if anything I would move the user's experience to its own section. The magic is in the server interactions.  The user's experience is intended to be so streamlined as to be almost irrelevant to the user.

Reply:
Oh, and from the user's experience, the registration process is the least important. Most important to highlight is the seamless workflow the user has when they sign in again.  This is one of the reasons why I suggest highlighting Safari in these examples, because it's a very beautiful experience.

So perhaps there should be 2 sections in this area: 1) Client-to-server Workflow, and 2) User Experience.